src/poetry/console/commands/config.py

```diff
@@ -138,6 +138,8 @@ def handle(self) -> int:
 
                 self.line(str(value))
             else:
+                if setting_key.split(".")[0] in self.LIST_PROHIBITED_SETTINGS:
+                    raise ValueError(f"Expected a value for {setting_key} setting.")
```

Comment:
Can we move this up, so we don't have to do the regex matching, which should not be relevant in this case?

Reply:
Yeah good point. It'll make the code cleaner. Fixed.